src/ConstructionSiteProvider.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { createContext, useState } from 'react';
+
+import { ConstructionSite } from './types';
+
+export const ConstructionSiteContext = createContext<{
+  constructionSites: ConstructionSite[];
+  setConstructionSites: React.Dispatch<React.SetStateAction<ConstructionSite[]>>;
+}>({ constructionSites: [], setConstructionSites: () => null });
+
+export const ConstructionSiteProvider = ({ children }: { children: React.ReactNode }) => {
+  const [constructionSites, setConstructionSites] = useState<ConstructionSite[]>([]);
+  return (
+    <ConstructionSiteContext.Provider value={{ constructionSites, setConstructionSites }}>
+      {children}
+    </ConstructionSiteContext.Provider>
```

Comment:
this provider is added in order to async query the data and rerender the widget, am i right?

Reply:
this provider is being added to not pass through the entire data objects as navigation params but instead to be able to work with identifiers/indices if needed.